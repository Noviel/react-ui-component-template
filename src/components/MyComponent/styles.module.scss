.main {
  background: darkorchid;
  border: 5px solid violet;
}
