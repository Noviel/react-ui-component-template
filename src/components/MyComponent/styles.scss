.main {
  background: darkorchid;
  border: 1px solid violet;
}
