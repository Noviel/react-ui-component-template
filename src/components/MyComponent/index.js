import React from 'react';

import styles from './styles.module.scss';

export default class Component extends React.Component {
  static defaultProps = {
    text: 'Hello',
  };

  render() {
    const { text } = this.props;
    return <div className={styles.main}>{text}</div>;
  }
}
