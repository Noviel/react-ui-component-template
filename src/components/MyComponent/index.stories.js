import React from 'react';
import { storiesOf } from '@storybook/react';

import MyComponent from './index';

storiesOf('MyComponent', module).add('default', () => <MyComponent />);
