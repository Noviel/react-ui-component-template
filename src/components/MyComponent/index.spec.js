import React from 'react';
import renderer from 'react-test-renderer';

import MyComponent from './index';

describe('MyComponent', () => {
  it('should render correct markup', () => {
    const component = renderer.create(
      <MyComponent />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
