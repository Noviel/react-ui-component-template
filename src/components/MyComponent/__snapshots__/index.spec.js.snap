// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correct markup 1`] = `
<div
  className="main"
>
  Hello
</div>
`;
